scripts/trigger_jenkins.py

```diff
@@ -0,0 +1,179 @@
+            if self.password is None:
+                raise JenkinsError("No API password provided")
+
+        # Figure out the jenkins URL
+        self.jenkins_url = jenkins_url
+        if self.jenkins_url is None:
+            # Try the environment variable
+            self.jenkins_url = os.getenv("JENKINS_URL")
+            if self.jenkins_url is None:
+                self.jenkins_url = self.JENKINS_URL
+
+        logger.debug('Using "%s" as jenkins URL', self.jenkins_url)
+
+        # Enable debugging in requests if needed.
+        if self.debug:
+            from http.client import HTTPConnection
+
+            HTTPConnection.debuglevel = 1
+            requests_log = logging.getLogger("urllib3")
+            requests_log.setLevel(logging.DEBUG)
+            requests_log.propagate = True
+
+        # Chain up to jenkins.Jenkins with URL and auth details
+        super(JenkinsAPI, self).__init__(
+            self.jenkins_url, self.user, self.password
+        )
+
+
+def main():
+    aparser = ArgumentParser(
+        description="Trigger Jenkins jobs through its API"
+    )
+    aparser.add_argument("job", metavar="JOB", help="jenkins job")
+    aparser.add_argument(
+        "params",
+        metavar="PARAM",
+        nargs="*",
+        help="job parameter in param=value format",
+    )
+    aparser.add_argument(
+        "--build-url",
+        action="store_true",
+        help="wait for and print Jenkins URL",
+    )
+    aparser.add_argument(
+        "--debug", action="store_true", help="enable HTTP debugging"
+    )
+    args = aparser.parse_args()
+
+    print(args.params)
+    # Create API connection
+    jenkins = JenkinsAPI(debug=args.debug)
+
+    # Build the parameters into a list of 2 member tuples
+    if len(args.params) > 0:
+        job_params = [tuple(param.split("=", 1)) for param in args.params]
+    else:
+        job_params = None
+
+    # Queue the job
+    try:
+        queue_number = jenkins.build_job(args.job, parameters=job_params)
+    except requests.HTTPError as err:
+        # If the job has parameters, Jenkins requires that you call it
+        # with buildWithParameters even if you don't want to override
+        # any of the defaults. Call again with an empty set of
+        # parameters so that build_job() is convinced to use the
+        # buildWithParameters Jenkins API.
+        if len(args.params) == 0 and err.response.status_code == 400:
+            job_params = [("", "")]
+            queue_number = jenkins.build_job(args.job, parameters=job_params)
+        else:
+            raise
+
+    # If the caller doesn't want the build URL, we're done
+    if not args.build_url:
+        return
+
+    # Poll the queue for a bit
+    waited = 0
+    while True:
+        data = jenkins.get_queue_item(queue_number)
+
+        if "executable" in data:
+            print(data["executable"]["url"])
+            break
+        elif data.get("cancelled", False) is True:
```

Comment:
```suggestion
        elif data.get("cancelled", False):
```

or is there some reason we specifically want to compare for identity with `True` rather than truthiness?

Reply:
Good question! This is a copy of something I wrote [several years ago](https://github.com/endlessm/eos-build/blob/master/ci-integration/jenkins-job-trigger), so I'd have to ask myself :face_with_spiral_eyes:. My guess is that I wasn't sure of the actual schema at the time and was worried that you'd get an empty string.